tests/phpunit/integration/Core/Admin/DashboardTest.php

```diff
@@ -70,4 +70,216 @@ function ( $cap ) {
 		$this->assertTrue( wp_script_is( 'googlesitekit-wp-dashboard', 'enqueued' ) );
 		$this->assertTrue( wp_style_is( 'googlesitekit-wp-dashboard-css', 'enqueued' ) );
 	}
+
+	/**
+	 * @group dashboard_widget
+	 */
+	public function test_render_googlesitekit_wp_dashboard_loading_container() {
+		$dashboard = new Dashboard( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$method    = new \ReflectionMethod( $dashboard, 'render_googlesitekit_wp_dashboard' );
+		$method->setAccessible( true );
+
+		ob_start();
+		$method->invoke( $dashboard );
+		$output = ob_get_clean();
```

Comment:
It would be preferable not to test private methods, which should ideally be treated as "black box" implementation details.

It's possible to retrieve the widget output by borrowing setup code from the test above (which could be lifted into a helper method), and retrieving the widget callback from `$wp_meta_boxes`:

```php
		global $wp_meta_boxes, $current_screen;
		// Clear out any registered meta boxes
		$wp_meta_boxes = array();
		// Set the current screen to the dashboard
		$current_screen = convert_to_screen( 'dashboard' );
		$admin_id       = $this->factory()->user->create( array( 'role' => 'administrator' ) );
		$dashboard      = new Dashboard( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );

		$dashboard->register();

		// Bypass user authentication requirement in map_meta_cap
		add_filter(
			'map_meta_cap',
			function ( $caps, $cap ) {
				if ( Permissions::VIEW_WP_DASHBOARD_WIDGET === $cap ) {
					$caps = array_filter(
						$caps,
						function ( $cap ) {
							return 'do_not_allow' !== $cap;
						}
					);
				}

				return $caps;
			},
			10,
			2
		);

		wp_set_current_user( $admin_id );

		require_once ABSPATH . 'wp-admin/includes/dashboard.php';

		wp_dashboard_setup();

		ob_start();

		$widget_callback = $wp_meta_boxes[ $current_screen->id ]['normal']['core']['google_dashboard_widget']['callback'];
		$widget_callback();
		$output = ob_get_clean();
```

It's a bit more code, but the setup can be shared between tests and it does allow us to avoid testing a private method. I'd suggest refactoring to use this approach.